From 8a9808b868c81d3fc65a84af74f0969dfc2df877 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 10 May 2016 02:00:11 -0700
Subject: [PATCH 3/4] clang: musl/ppc does not support 128-bit long double

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 lib/Basic/Targets.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/Basic/Targets.cpp b/lib/Basic/Targets.cpp
index 5e9ea68..33f0d89 100644
--- a/lib/Basic/Targets.cpp
+++ b/lib/Basic/Targets.cpp
@@ -1592,7 +1592,8 @@ public:
       break;
     }
 
-    if (getTriple().getOS() == llvm::Triple::FreeBSD) {
+    if (getTriple().getOS() == llvm::Triple::FreeBSD
+        || getTriple().isMusl()) {
       LongDoubleWidth = LongDoubleAlign = 64;
       LongDoubleFormat = &llvm::APFloat::IEEEdouble;
     }
-- 
2.9.0

