From 0f183864a6e89461143e082d4b7e5df4cd0f4480 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 10 May 2016 02:00:11 -0700
Subject: [PATCH 03/15] clang: musl/ppc does not support 128-bit long double

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 clang/lib/Basic/Targets/PPC.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/clang/lib/Basic/Targets/PPC.h b/clang/lib/Basic/Targets/PPC.h
index 058970a0e09..ba5b941bc08 100644
--- a/clang/lib/Basic/Targets/PPC.h
+++ b/clang/lib/Basic/Targets/PPC.h
@@ -331,7 +331,8 @@ public:
       break;
     }
 
-    if (getTriple().isOSFreeBSD()) {
+    if (getTriple().isOSFreeBSD()
+        || getTriple().isMusl()) {
       LongDoubleWidth = LongDoubleAlign = 64;
       LongDoubleFormat = &llvm::APFloat::IEEEdouble();
     }
-- 
2.20.1

