From 082dd1f20b838510dca1481e9479c57ce8d03dff Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 10 May 2016 02:00:11 -0700
Subject: [PATCH] clang: musl/ppc does not support 128-bit long double

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 clang/lib/Basic/Targets/PPC.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/clang/lib/Basic/Targets/PPC.h b/clang/lib/Basic/Targets/PPC.h
index 2ceb0b0cbf1c..5b9a47d02f1a 100644
--- a/clang/lib/Basic/Targets/PPC.h
+++ b/clang/lib/Basic/Targets/PPC.h
@@ -407,6 +407,10 @@ public:
       LongDoubleFormat = &llvm::APFloat::IEEEdouble();
     }
 
+    if (getTriple().isMusl()) {
+      LongDoubleWidth = LongDoubleAlign = 64;
+      LongDoubleFormat = &llvm::APFloat::IEEEdouble();
+    }
     // PPC32 supports atomics up to 4 bytes.
     MaxAtomicPromoteWidth = MaxAtomicInlineWidth = 32;
   }
