--- a/source3/wscript_build	2019-03-14 09:03:29.498617250 +0100
+++ b/source3/wscript_build	2019-03-14 09:06:19.824949694 +0100
@@ -124,7 +124,6 @@
                      '!get_attr_list']
 private_pdb_match.append('!pdb_nds_*')
 private_pdb_match.append('!pdb_ldapsam_init*')
-private_pdb_match.append('!pdb_*_init')
 private_pdb_match = private_pdb_match + ldapsam_pdb_match
 
 bld.SAMBA3_LIBRARY('samba-passdb',
